lms/views/helpers/_via.py

```diff
@@ -18,28 +92,9 @@ def via_url(request, document_url):
     :return: A URL string
     """
 
-    # Default via parameters
-    options = {
-        "via.client.openSidebar": "1",
-        "via.client.requestConfigFromFrame.origin": request.host_url,
-        "via.client.requestConfigFromFrame.ancestorLevel": "2",
-    }
-
-    if request.feature("use_legacy_via"):
-        return _legacy_via_url(
-            request.registry.settings["legacy_via_url"], document_url, options
-        )
-
-    options["url"] = document_url
-    via_service_url = urlparse(request.registry.settings["via_url"])
-
-    return via_service_url._replace(path="/route", query=urlencode(options)).geturl()
-
-
-def _legacy_via_url(via_service_url, document_url, options):
-    parsed_url = urlparse(document_url)
-
-    params = [kv for kv in parse_qsl(parsed_url.query) if not kv[0].startswith("via.")]
-    params.extend(options.items())
-
-    return via_service_url + parsed_url._replace(query=urlencode(params)).geturl()
+    return _ViaClient(
+        service_url=request.registry.settings["via_url"],
+        legacy_service_url=request.registry.settings["legacy_via_url"],
+        host_url=request.host_url,
+        legacy_mode=request.feature("use_legacy_via"),
+    ).url_for(_ViaDoc(document_url))
```

Comment:
This is the opportunity to construct the doc with `content_type` and have that be passed in to `via_url()`. Hopefully we can then set that when using a Canvas doc or the like.